api/src/main/java/javax/observer/collection/CollectionObservable.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package javax.observer.collection;
+
+import java.util.Collection;
+import javax.observer.Subscription;
+
+/**
+ * 
+ * @param <T> element type of the collection
+ * @param <OC> observed collection type
+ * @param <CC> collection change type
+ * 
+ * @author Michael Hoffer (info@michaelhoffer.de)
+ */
+public interface CollectionObservable<T, OC extends Collection<T>, CC extends CollectionChange<T>> {
```

Comment:
In general I don't like observer patterns where the observable is forced to be the object that is observed. In many cases it is. But not always. 

Sometimes it is more efficient or elegant to have only one observable for many collections. A listener subscribes to the changes of all of them. I sometimes use this pattern for hierarchical data structures. Subscribing to all collections individually is just overkill.